File: Gerontocracy.Core/BusinessObjects/Board/Post.cs
using System;
using System.Collections.Generic;

namespace Gerontocracy.Core.BusinessObjects.Board
{
    public class Post
    {
        public long Id { get; set; }
        public string Content { get; set; }
        public DateTime CreatedOn { get; set; }
        
        public long? UserId { get; set; }
        public string UserName { get; set; }
        
        public int Likes { get; set; }
        public int Dislikes { get; set; }

        public LikeType? UserLike { get; set; }

        public long? ParentId { get; set; }

        public List<Post> Children { get; set; }
    }
}

File: Gerontocracy.Web/src/gerontocracy/modules/board/models/post.ts
import { LikeType } from './like-type';

export interface Post {
    id: number;
    content: string;
    createdOn: Date;
    userId: number;
    userName: string;
    userLike: LikeType;
    likes: number;
    dislikes: number;
    children: Post[];
}
